### ISSUE TITLE
Better docs please :)

### ISSUE THREAD

**Issue by kristianmandrup** [htm-community/comportex#45]
I've just been reading through:
http://floybix.github.io/2014/11/05/htm-protocols

Pretty hard to grasp from an "outsider". I'd suggest visualizing the architecture, with some kind of graph to show where each kind of Protocol goes in the IO flow... I'd love to help out if I could.
Or maybe just link to some more/better articles/videos on htm theory and how it fits in with data structures and info flows

**Comment by floybix** [comment]
I agree the docs are woeful. I've been reluctant to put effort into docs since I still think of the software as in flux (and not useful in practice right now).

Speaking of which, I have recently starting planning to get rid of "regions" altogether. Just leaving layers, which could be assembled in different ways.

**Comment by floybix** [comment]
If it helps, currently regions are basically containers for layers, but unless you are running exotic experiments you will have just one layer in a region, and it will be known as "layer-3" (neuroscience term).

**Comment by kristianmandrup** [comment]
"I have recently starting planning to get rid of `regions` altogether. Just leaving layers, which could be assembled in different ways."

That sounds very reasonable on the face of it. Better with simpler constructs/models that can be used as generic building blocks I think :)

**Comment by kristianmandrup** [comment]
Can I make a "brain" with layers of any depth I like using this model? I recall that the latest winners of Image recognition competition used more than a hundred layers...

**Comment by floybix** [comment]
https://discourse.numenta.org/t/h-is-for-hierarchy-but-where-is-it/333

TLDR: this is a research topic, it is not useful for anything right now...

**Comment by kristianmandrup** [comment]
"A _region_ is the main composable unit in this library. It represents a bank of neurons arranged in columns, responding to an array of feed-forward input bits, as well as distal connections to
itself and possibly other regions."

So removing regions would be a major change!! Would be nice with some docs on how to compose the brain architecture...
